src/include/btree_inline.h

```diff
@@ -1545,7 +1545,8 @@ __wt_ref_addr_copy(WT_SESSION_IMPL *session, WT_REF *ref, WT_ADDR_COPY *copy)
      * WT_ADDRs and swapped into place. The content of the two WT_ADDRs are identical, and we don't
      * care which version we get as long as we don't mix-and-match the two.
      */
-    WT_ACQUIRE_READ_WITH_BARRIER(addr, (WT_ADDR *)ref->addr);
+    addr = (WT_ADDR *)ref->addr;
+    WT_ACQUIRE_BARRIER();
```

Comment:
What happened here?

Reply:
The compiler isn't happy with taking `ref->addr` address after we cast it to a struct pointer:
`error: cannot take the address of an rvalue of type 'WT_ADDR *' (aka 'struct __wt_addr *')`
Dropping the typecast also causes compiler issues since `ref->addr` is a void pointer.

I can dig further if needed, but for now I've left this alone since `ref->addr` hasn't been flagged by TSan. 
There was a separate field where `WT_ACQUIRE_READ_WITH_BARRIER` needed the atomics, and this is a flow on effect of adding atomic_load to `WT_ACQUIRE_READ_WITH_BARRIER`.